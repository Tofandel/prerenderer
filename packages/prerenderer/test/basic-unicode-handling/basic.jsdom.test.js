const path = require('path')
const Prerenderer = require('../../')
const Renderer = require('../../../renderer-jsdom')

const EXPECTED_HTML = '<!DOCTYPE html><html><head>\n  <title>Prerenderer Test</title>\n</head>\n<body>\n  <script>\n    document.addEventListener(\'DOMContentLoaded\', () => {\n      document.body.innerHTML += \'<p>Render Output</p>\'\n    })\n  </script>\n\n\n<p>Render Output</p></body></html>'

test('renders 1 route', async () => {
  const expectedResult = [{
    originalRoute: '/tést.html',
    route: '/tést.html',
    html: EXPECTED_HTML
  }]

  const prerenderer = new Prerenderer({
    staticDir: path.resolve(__dirname),
    renderer: new Renderer()
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/tést.html'])
  console.log(renderedRoutes)
  prerenderer.destroy()
  expect(renderedRoutes).toEqual(expectedResult)
})
