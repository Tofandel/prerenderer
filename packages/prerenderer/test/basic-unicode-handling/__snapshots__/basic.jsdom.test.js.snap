// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders route with special character in path 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      document.body.innerHTML += '<p>Render Output</p>'
    })
  </script>


<p>Render Output</p></body></html>",
    "originalRoute": "/tést.html",
    "route": "/tést.html",
  },
]
`;
